controllers/services/auth/auth_controller_actions.go

```diff
@@ -81,84 +81,128 @@ func copyGroups(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 		return client.IgnoreNotFound(err)
 	}
 	foundGroups, found, _ := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
-	if !found {
-		return errors.New("no groupsConfig found in dashboardConfig")
-	}
+	if found {
+		added := common.AddMissing(&ai.Spec.AdminGroups, foundGroups["adminGroups"])
+		added += common.AddMissing(&ai.Spec.AllowedGroups, foundGroups["allowedGroups"])
 
-	added := common.AddMissing(&ai.Spec.AdminGroups, foundGroups["adminGroups"])
-	added += common.AddMissing(&ai.Spec.AllowedGroups, foundGroups["allowedGroups"])
+		// only update if we found a new group in the list
+		if added == 0 {
+			return nil
+		}
+
+		err = rr.Client.Update(ctx, ai)
+		if err != nil {
+			return errors.New("error adding groups to Auth CR")
+		}
+	}
 
-	if added == 0 {
+	if !found {
 		return nil
 	}
 
-	// only update if we found a new group in the list
-	err = rr.Client.Update(ctx, ai)
+	unstructured.RemoveNestedField(odhObject.Object, "spec", "groupsConfig")
+	err = rr.Client.Update(ctx, odhObject)
 	if err != nil {
-		return errors.New("error adding groups to Auth CR")
+		return errors.New("error removing groupsConfig from odhDashboardConfig")
 	}
 
 	return nil
 }
 
 func bindRole(ctx context.Context, rr *odhtypes.ReconciliationRequest, groups []string, roleBindingName string, roleName string) error {
-	groupsToBind := []rbacv1.Subject{}
-	for _, e := range groups {
-		// we want to disallow adding system:authenticated to the adminGroups
-		if roleName == "admingroup-role" && e == "system:authenticated" {
-			log := logf.FromContext(ctx)
-			log.Info("system:authenticated cannot be added to adminGroups")
-			continue
-		}
-		rs := rbacv1.Subject{
-			Kind:     "Group",
-			APIGroup: "rbac.authorization.k8s.io",
-			Name:     e,
+	var rb *rbacv1.RoleBinding
```

Comment:
I think I can simplify this and the cluster binding function now that I look at it by defaulting to the empty subjects list.